src/libs/polycube/src/table.cpp

```diff
@@ -36,13 +37,23 @@ class RawTable::impl {
   int first(void *key);
   int next(const void *key, void *next);
 
+  int pop(void *value);
+  int push(const void *value);
+
  private:
+  bool is_queue_stack_enabled;
+  bool is_batch_enabled;
   int fd_;
 };
 
-RawTable::impl::~impl() {}
+RawTable::impl::~impl() {
+}
 
-RawTable::impl::impl(void *op) : fd_(*(int *)op) {}
+RawTable::impl::impl(void *op) : fd_(*(int *)op) {
+  auto version = get_kernel_release();
+  is_queue_stack_enabled = compare_kernel_release(version, QUEUE_STACK_KERNEL_RELEASE);
+  is_batch_enabled = compare_kernel_release(version, BATCH_KERNEL_RELEASE);
```

Comment:
Did you add the batch support? If not, maybe it is better to add this change in that PR.

Reply:
I haven't added it yet, but I tried to make it work in a personal branch, but I'm still figuring out how to correctly call the function (unfortunately there's no doc). Should I implement it in this PR?

Reply:
Solved in 523ff8b 
Unfortunately, batch works only with primitive data types and with standard maps (not queue/stack). So I've added these new function to the RawTable class that can be used, but in Dynmon I've commented the code where I performed batch research since no useful by now

Reply:
Maybe in the following kernel releases they will make it work, and then we will already have the code for the new extraction type :)